src/bhv_basic_move.cpp

```diff
@@ -105,13 +107,21 @@ Bhv_BasicMove::execute( PlayerAgent * agent )
     agent->debugClient().setTarget( target_point );
     agent->debugClient().addCircle( target_point, dist_thr );
 
-    if ( ! Body_GoToPoint( target_point, dist_thr, dash_power
-                           ).execute( agent ) )
-    {
-        Body_TurnToBall().execute( agent );
+
+    if ( Body_GoToPoint( target_point, dist_thr, dash_power
+			 ).execute( agent ) ||
+	 Body_TurnToBall().execute( agent ) ) {
+      if (ball.rposValid() || wm.self().collidesWithPost()) {
+	success = true;
+      } else {
```

Comment:
Lets remove both of these else clauses. Success will be initialized as false (as defined on line-110) and these expression will only make it true if the conditions are met.